hyperspy/misc/material.py

```diff
@@ -253,6 +256,209 @@ def density_of_mixture_of_pure_elements(weight_percent,
+        for i, energy in enumerate(energies):
+            if isinstance(energy, str):
+                energies[i] = utils_eds._get_energy_xray_line(energy)
+    index = np.searchsorted(energies_db, energies)
+    mac_res = np.exp(np.log(macs[index - 1]) +
+                     np.log(macs[index] / macs[index - 1]) *
+                     (np.log(energies / energies_db[index - 1]) /
+                     np.log(energies_db[index] / energies_db[index - 1])))
+    return np.nan_to_num(mac_res)
+
+
+def _mass_absorption_coefficient_of_mixture_of_pure_elements(weight_percent,
+                                                             elements,
+                                                             energies):
+    """Calculate the mass absorption coefficient for X-ray absorbed in a
+    mixture of elements.
+
+    The mass absorption coefficient is calculated as a weighted mean of the
+    weight percent and is retrieved from the database of Chantler2005.
+
+    Parameters
+    ----------
+    weight_percent: np.array
+        The composition of the absorber(s) in weight percent. The first
+        dimension of the matrix corresponds to the elements.
+    elements: list of str
+        The list of element symbol of the absorber, e.g. ['Al','Zn'].
+    energies: float or list of float or str or list of str
+        The energy or energies of the X-ray in keV, or the name of the X-rays,
+        e.g. 'Al_Ka'.
+
+    Examples
+    --------
+    >>> hs.material.mass_absorption_coefficient_of_mixture_of_pure_elements(
+    >>>     elements=['Al','Zn'], weight_percent=[50,50], energies='Al_Ka')
+    2587.4161643905127
+
+    Return
+    ------
+    float or array of float
+    mass absorption coefficient(s) in cm^2/g
+
+    See also
+    --------
+    hs.material.mass_absorption_coefficient
+
+    Note
+    ----
+    See http://physics.nist.gov/ffast
+    Chantler, C.T., Olsen, K., Dragoset, R.A., Kishore, A.R., Kotochigova,
+    S.A., and Zucker, D.S. (2005), X-Ray Form Factor, Attenuation and
+    Scattering Tables (version 2.1).
+    """
+    if len(elements) != len(weight_percent):
+        raise ValueError(
+            "Elements and weight_fraction should have the same length")
+    if hasattr(weight_percent[0], '__iter__'):
+        weight_fraction = np.array(weight_percent)
+        weight_fraction /= np.sum(weight_fraction, 0)
+        mac_res = np.zeros([len(energies)]+list(weight_fraction.shape[1:]))
+        for element, weight in zip(elements, weight_fraction):
+            mac_re = mass_absorption_coefficient(element, energies)
+            mac_res += np.array([weight * ma for ma in mac_re])
+        return mac_res
+    else:
+        mac_res = np.array([mass_absorption_coefficient(
+            el, energies) for el in elements])
+        mac_res = np.dot(weight_percent, mac_res) / np.sum(weight_percent, 0)
+        return mac_res
+
+
+def mass_absorption_coefficient_of_mixture_of_pure_elements(weight_percent,
```

Comment:
I'm all for self-documenting code, but with a style guide saying a maximum line length of 80 characters, this function name is way too long. How about `mass_absorption_mixture`?